src/lib/y2storage/devicegraph_sanitizer.rb

```diff
@@ -17,289 +17,71 @@
   class DevicegraphSanitizer
-    # @return [Devicegraph]
-    attr_reader :devicegraph
-
     # Constructor
     #
     # @param devicegraph [Devicegraph] devicegraph to sanitize
     def initialize(devicegraph)
       @devicegraph = devicegraph
     end
 
-    # Errors that need to be fixed in order to obtain a sanitized devicegraph
-    #
-    # @return [Array<DevicegraphSanitizer::Error>] empty if the devicegraph is already sanitized.
-    def errors
-      @errors ||= errors_for(devicegraph)
-    end
-
     # Sanitized version of the devicegraph
     #
-    # @return [Y2Storage::Devicegraph]
+    # @return [Devicegraph]
     def sanitized_devicegraph
-      @sanitized_devicegraph ||= sanitize(devicegraph.dup)
+      @sanitized_devicegraph ||= sanitize
     end
 
     private
 
-    # Errors in the given devicegraph
-    #
-    # @param devicegraph [Y2Storage::Devicegraph]
-    # @return [Array<DevicegraphSanitizer::Error>]
-    def errors_for(devicegraph)
-      lvm_vgs_errors(devicegraph) +
-        bcaches_errors(devicegraph) +
-        filesystems_errors(devicegraph)
-    end
-
-    # Errors related to LVM VGs in the given devicegraph
-    #
-    # @param devicegraph [Y2Storage::Devicegraph]
-    # @return [Array<DevicegraphSanitizer::Error>]
-    def lvm_vgs_errors(devicegraph)
-      devicegraph.lvm_vgs.flat_map { |v| lvm_vg_errors(v) }.compact
-    end
-
-    # Errors for an LVM VG
-    #
-    # @param vg [Y2Storage::LvmVg]
-    # @return [Array<DevicegraphSanitizer::Error>]
-    def lvm_vg_errors(vg)
-      errors = []
-
-      errors << MissingLvmPvError.new(vg) if MissingLvmPvError.check(vg)
-
-      errors
-    end
-
-    # Errors related to Bcache in the given devicegraph
-    #
-    # @param devicegraph [Y2Storage::Devicegraph]
-    # @return [Array<DevicegraphSanitizer::Error>]
-    def bcaches_errors(devicegraph)
-      errors = []
-
-      errors << UnsupportedBcacheError.new if UnsupportedBcacheError.check(devicegraph)
-
-      errors
-    end
-
-    # Errors related to filesystems in the given devicegraph
-    #
-    # @param devicegraph [Y2Storage::Devicegraph]
-    # @return [Array<DevicegraphSanitizer::Error>]
-    def filesystems_errors(devicegraph)
-      devicegraph.filesystems.flat_map { |f| filesystem_errors(f) }.compact
-    end
+    # @return [Devicegraph]
+    attr_reader :devicegraph
 
-    # Errors for a filesystem
+    # Generates a sanitized copy of the devicegraph
     #
-    # @param filesystem [Y2Storage::Filesystems::Base]
-    # @return [Array<DevicegraphSanitizer::Error>]
-    def filesystem_errors(filesystem)
-      errors = []
+    # @return [Devicegraph]
+    def sanitize
+      sanitized = devicegraph.dup
 
-      errors << InactiveRootError.new(filesystem) if InactiveRootError.check(filesystem)
+      issues.each { |i| fix_issue(sanitized, i) }
 
-      errors
+      sanitized
     end
 
-    # Sanitizes a given devicegraph
+    # Fixes the error (if possible)
```

Comment:
I'd say _Fixes the issue, when possible_. But right now I a bit lost about when we should call them error and when issues.

Reply:
We have not differenciated it too much yet. Actually, an issue can be an error, warning, etc. By default, all the issues are generated as warnings. Till now, the category is not that important. Maybe we need to define what an "error" means for us (do not allow to continue and foce to abort?).